File: devops_scripts/extract_mappings.php
<?php
$inputFile = "d:/NTT_WEBSITE/backups/database_backup.sql";
$outputFile = "d:/NTT_WEBSITE/full_mappings_base.sql";
$targetTables = ["categories", "tags", "post_categories", "post_tags", "medias"];
$handle = fopen($inputFile, "r");
$out = fopen($outputFile, "w");
if ($handle && $out) {
    while (($line = fgets($handle)) !== false) {
        foreach ($targetTables as $table) {
            if (strpos($line, "INSERT INTO `$table`") !== false) {
                fwrite($out, $line);
                break;
            }
        }
    }
    fclose($handle);
    fclose($out);
    echo "EXTRACTION_COMPLETE";
} else { echo "FILE_OPEN_FAILED"; }
?>

File: devops_scripts/extract_mappings.py

import re

input_file = r'd:\NTT_WEBSITE\backups\database_backup.sql'
output_file = r'd:\NTT_WEBSITE\full_mappings_base.sql'

target_tables = ['categories', 'tags', 'post_categories', 'post_tags', 'medias']
pattern = re.compile(r"INSERT INTO `(" + "|".join(target_tables) + r")`")

with open(input_file, 'r', encoding='utf-8', errors='ignore') as infile:
    with open(output_file, 'w', encoding='utf-8') as outfile:
        for line in infile:
            if pattern.search(line):
                outfile.write(line)

print(f"Extraction complete to {output_file}")
